Tasks/MavenAuthenticateV0/mavenutils.ts

```diff
@@ -0,0 +1,200 @@
+        }
+    }   
+
+    return serverElements;
+}
+
+export function readXmlFileAsJson(filePath: string): Q.Promise<any> {
+    return readFile(filePath, 'utf-8')
+        .then(convertXmlStringToJson);
+}
+
+export function writeJsonAsSettingsFile(filePath: string, jsonContent: any): Q.Promise<void> {
+    return writeJsonAsXmlFile(filePath, jsonContent.settings, 'settings');
+}
+
+export function mavenSettingsJsonInsertServer (json: any, serverJson:any): any {
+    if (!json) {
+        json = {};
+    }
+    if (!json.settings || typeof json.settings === "string") {
+        json.settings = {};
+    }
+    if (!json.settings.$) {
+        json.settings.$ = {};
+        json.settings.$['xmlns'] = 'http://maven.apache.org/SETTINGS/1.0.0';
+        json.settings.$['xmlns:xsi'] = 'http://www.w3.org/2001/XMLSchema-instance';
+        json.settings.$['xsi:schemaLocation'] = 'http://maven.apache.org/SETTINGS/1.0.0' + os.EOL + 'https://maven.apache.org/xsd/settings-1.0.0.xsd';
+    }
+    if (!json.settings.servers) {
+        json.settings.servers = {};
+    }
+    addPropToJson(json.settings.servers, 'server', serverJson);
+    return json;
+}
+
+
+function addPropToJson(obj: any, propName:string, value: any): void {
+    if (!obj) {
+        obj = {};
+    }
+
+    if (obj instanceof Array) {
+        let propNode = obj.find(o => o[propName]);
+        if (propNode) {
+            obj = propNode;
+        }
+    }
+
+    let containsId: (o) => boolean = function(o) {
+        if (value && value.id) {
+            if (o.id instanceof Array) {
+                return o.id.find((v) => {
+                    return v === value.id;
+                });
+            } else {
+                return value.id === o.id;
+            }
+        }
+        return false;
+    };
+
+    if (propName in obj) {
+        if (obj[propName] instanceof Array) {
+            let existing = obj[propName].find(containsId);
+            if (existing) {
+                tl.warning(tl.loc('Warning_FeedEntryAlreadyExists', value.id));
+                tl.debug('Entry: ' + value.id);
+            } else {
+                obj[propName].push(value);
+            }
+        } else if (typeof obj[propName] !== 'object') {
+            obj[propName] = [obj[propName], value];
+        } else {
+            let prop = {};
+            prop[propName] = value;
+            obj[propName] = [obj[propName], value];
+        }
+    } else if (obj instanceof Array) {
+        let existing = obj.find(containsId);
+        if (existing) {
+            tl.warning(tl.loc('Warning_FeedEntryAlreadyExists', value.id));
+            tl.debug('Entry: ' + value.id);
+        } else {
+            let prop = {};
+            prop[propName] = value;
+            obj.push(prop);
+        }
+    } else {
+        obj[propName] = value;
+    }
+}
+
+
+function writeJsonAsXmlFile(filePath: string, jsonContent: any, rootName:string): Q.Promise<void> {
```

Comment:
nit: may be rename to jsonToXmlFileConverter?

Reply:
Fixed